algorithms/unit_tests/TestStdAlgorithmsCompileOnly.cpp

```diff
@@ -507,6 +507,24 @@ struct TestStruct {
   }
 };
 
+#ifndef KOKKOS_ENABLE_CXX17
+template <typename ViewType>
+constexpr bool
+test_kokkos_iterator_satify_std_random_access_iterator_concept() {
+  return std::random_access_iterator<
```

Comment:
Where do we get the `<concepts>` header from?  Is it guaranteed that it is always transitively included?

Reply:
According to https://en.cppreference.com/w/cpp/iterator/random_access_iterator, `random_access_iterator` comes from `<iterator>` and we include that header in `Kokkos_RandomAccessIterator.hpp` which is included in `Kokkos_StdAlgorithms.hpp` that we include at the top of the file. 
In short, `RandomAccessIterator` needs the `iterator` header and so `random_access_iterator` is available whenever a definition of `RandomAccessIterator` is needed.